engine/src/main/java/org/terasology/rendering/nui/contextMenu/ContextMenuTree.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.contextMenu;
+
+import com.google.common.collect.Maps;
+
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class ContextMenuTree {
+    private Map<String, ContextMenuOption> options = Maps.newHashMap();
+    private Map<String, ContextMenuTree> submenues = Maps.newHashMap();
```

Comment:
What if the user wants to specify the order of the menu items?

I would go for a `List<AbstractContextMenuItem>`.